### ISSUE TITLE
[FEATURE REQ] Controller Profile Certification Records

### ISSUE THREAD

**Issue by braukStauter** [zabartcc/ui#58]
## Describe the solution you'd like
A transition from GRP "Major" and "Minor" certification records contained within ZAB controller profiles (see fig. 1), to a system which contains Tier 1 and Tier 2 endorsements for fields and certification levels for which the controller is rated (see fig. 2, proposed) and generally increases readability of controller certification within controller profiles.

![image](https://github.com/zabartcc/issue-reporting/assets/24839808/ae1584be-3ea8-46bb-a07b-da5e33d93977)
_Fig 1_ ↑ 

![image](https://github.com/zabartcc/issue-reporting/assets/24839808/948ccb75-f0bd-4d19-8fc3-ef17e911287e)
_Fig 2_ ↑ 

## Describe alternatives you've considered
N/A.

## Is your feature request related to a problem? Please describe.
This feature request is in response to global VATSIM transition from GRP to GCAP effective March 1st, 2024.

## Additional context
N/A.

**Comment by cdconn00** [comment]
Triaged and put on the backlog.

**Comment by cdconn00** [comment]
@braukStauter I've started on this rework. Planning to leave the "Controller Certification" section off, as this information is already displayed on the page and cannot be edited.

**Comment by braukStauter** [comment]
Also now that I’m thinking about it, you’ll have to add KPHX GND, TWR, and
P50 as well to differentiate

**Comment by cdconn00** [comment]
Okay, thanks.

**Comment by cdconn00** [comment]
Should there be a center button for c1+ visitors who have not yet completed a ctr checkout? @braukStauter

**Comment by braukStauter** [comment]
Sure - so let’s do:

Endorsements:
Tier 1:
PHX GND
PHX TWR
PHX P50

Tier 2:
ABQ
FLG
LUF
SAF

Visitors:
ZAB Enroute

**Comment by cdconn00** [comment]
Thanks. Sorry to be edge case eddie -- if it's a new C1 transfer who just came in... 

Maybe makes sense to leave ZAB Enroute in Tier One?

Spitballing, up to you

**Comment by braukStauter** [comment]
I mean, realistically a visiting C1 isn’t able to control ZAB until they
have all of the endorsements anyway, so we might not even need to use it.

But for the sake of clarity we probably should. You can throw it under Tier
1 that’s fine with me!

**Comment by cdconn00** [comment]
Status update: being reviewed by ARTCC staff in the beta environment before code review.

**Comment by cdconn00** [comment]
To apply this logic use the following script to ensure you map existing certs.

```
import User from './models/User.js';
import dotenv from 'dotenv';
import mongoose from 'mongoose';

dotenv.config();
mongoose.set('toJSON', {virtuals: true});
mongoose.set('toObject', {virtuals: true});
mongoose.connect(); // Fill out connection string.
const db = mongoose.connection;
db.once('open', () => console.log('Successfully connected to MongoDB'));

async function main() {
    const oldMinorCodes = ['app', 'twr', 'gnd'];
    const newMinorCodes = ['kluf', 'ksaf', 'kabq', 'kflg']

    var users = await User.find({deleted: false}).select('-email -idsToken -discordInfo').sort({
        rating: 'desc',
        lname: 'asc',
        fname: 'asc'
    });

    users.forEach(async (user) => {
        let newCertsToAdd = [];
        
        if (user.certCodes.includes('zab'))
        {
            newCertsToAdd.push('enroute')
        }

        if (user.certCodes.includes('p50app'))
        {
            newCertsToAdd.push('p50')
        }

        if (user.certCodes.includes('p50twr'))
        {
            newCertsToAdd.push('kphxtower')
        }    

        if (user.certCodes.includes('p50gnd'))
        {
            newCertsToAdd.push('kphxground')
        }
        
        if (user.certCodes.some(c => oldMinorCodes.includes(c))){
            newCertsToAdd = newCertsToAdd.concat(newMinorCodes);
        }

        user.certCodes = user.certCodes.concat(newCertsToAdd);
        user.certCodes = [...new Set(user.certCodes)] // Make sure roles are unique.

        await user.save();
    });
}

main();
```

**Comment by cdconn00** [comment]
Closed by #71 @braukStauter
